File: backend/src/main/java/com/library/library_management/model/Book.java
package com.library.library_management.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "books")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String author;
    private String isbn;
    private String genre;

    @Column(length = 2000)
    private String synopsis;

    private String coverUrl;
    private Integer totalCopies;
    private Integer availableCopies;

    private Integer publicationYear;

    @Builder.Default
    @ElementCollection
    @CollectionTable(name = "book_tags", joinColumns = @JoinColumn(name = "book_id"))
    @Column(name = "tag")
    private java.util.Set<String> tags = new java.util.HashSet<>();
}

File: backend-node/src/entities/Book.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from "typeorm";
import { BookTag } from "./BookTag";

@Entity("books")
export class Book {
    @PrimaryGeneratedColumn({ type: "bigint" })
    id: number;

    @Column()
    title: string;

    @Column()
    author: string;

    @Column()
    isbn: string;

    @Column()
    genre: string;

    @Column({ length: 2000, nullable: true })
    synopsis: string;

    @Column({ name: "cover_url", nullable: true })
    coverUrl: string;

    @Column({ name: "total_copies", nullable: true })
    totalCopies: number;

    @Column({ name: "available_copies", nullable: true })
    availableCopies: number;

    @Column({ name: "publication_year", nullable: true })
    publicationYear: number;

    @OneToMany(() => BookTag, (tag) => tag.book, { cascade: true, eager: true })
    tags: BookTag[];
}
